lib/infra_conversion_throttler.rb

```diff
@@ -19,4 +19,43 @@ def self.pending_conversion_jobs
     _log.info("Pending InfraConversionJob: #{pending.count}")
     pending.group_by { |job| job.migration_task.destination_ems }
   end
+
+  # @return [Hash] the list of jobs in state 'running', grouped by conversion host
+  def self.running_conversion_jobs
+    running = InfraConversionJob.where(:state => 'running')
+    _log.info("Running InfraConversionJob: #{running.count}")
+    running.group_by { |job| job.migration_task.conversion_host }
+  end
+
+  # Calculate and apply the limits for all running jobs.
+  # The supported limits are:
+  #   - CPU per conversion host
+  #   - network per conversion host
+  #
+  # The limits can be retrieved per conversion host or from the default setting.
+  # When virt-v2v-wrapper starts the task stores the throttling file path in task.options[:virt-v2v-wrapper]['throttling_file'].
+  # There is no need to adjust limits if virt-v2v-wrapper has not been called by the task.
+  # The resources are evenly split between the jobs on a same conversion host.
+  # There is no need to adjust limits if they have not changed.
+  # Applying the limits is done via the conversion_host which handles the writing.
+  def self.apply_limits
+    running_conversion_jobs.each do |ch, jobs|
+      number_of_jobs = ch.active_tasks.size
```

Comment:
Hm if ConversionHost#active_tasks got out of sync with the running jobs somehow and was 0 then [this division](https://github.com/ManageIQ/manageiq/pull/18576/files#diff-8dffd85ef5e1772dce26180aa8ff6d65R51) will blow up...maybe set this to 1 if it is 0?

Reply:
Good catch. `jobs.size` is safer because `jobs` can't be empty inside the loop.